### PR TITLE
time_weighted_retriever: use a timestamp if needed

### LINKED ISSUE #3685: TimeWeightedVectorStoreRetriever (TWVSR) and ChromaDb  vector store  - base.py
Hi there. 

Realise there is a lot happening and this this looks to be something that has been missed. When trying to use TWVSR with ChromaDb it errors because of the lack of implementation of the following, namely __similarity_search_with_relevance_scores, inside of base.py

    `def _similarity_search_with_relevance_scores(
        self,
        query: str,
        k: int = 4,
        **kwargs: Any,
    ) -> List[Tuple[Document, float]]:
        """Return docs and relevance scores, normalized on a scale from 0 to 1.
        0 is dissimilar, 1 is most similar.
        """
        raise NotImplementedError`

Trying to make a work-around now
Many thanks
Ian

### PR DESCRIPTION
**Pull request by flepied**
If last_accessed_at metadata is a float use it as a timestamp. This allows to support vector stores that do not store datetime objects like ChromaDb.

Fixes: https://github.com/langchain-ai/langchain/issues/3685

### CODE REVIEW

File: libs/langchain/Makefile [hunk 1]
```diff
@@ -44,23 +44,20 @@ api_docs_linkcheck:
 # Define a variable for the test file path.
 TEST_FILE ?= tests/unit_tests/
 
-test:
-	poetry run pytest --disable-socket --allow-unix-socket $(TEST_FILE)
```
Comment (baskaryan) [inline libs/langchain/Makefile:48]: why the change?

Comment (flepied) [reply libs/langchain/Makefile:48]: it was to have `make test` pass on my system as the tests are locale dependent.  I can remove this part.

Comment (flepied) [reply libs/langchain/Makefile]: I removed this part

**Review by hwchase17 — APPROVED** [review]

File: libs/langchain/langchain/retrievers/time_weighted_retriever.py [hunk 2]
```diff
@@ -47,6 +47,14 @@ class Config:
 
         arbitrary_types_allowed = True
 
+    def _document_get_date(self, field: str, document: Document) -> datetime.datetime:
+        """Return the value of the date field of a document."""
+        if field in document.metadata:
+            if type(document.metadata[field]) == float:
```
Comment (chmouel) [inline libs/langchain/langchain/retrievers/time_weighted_retriever.py:53]: what about using `isinstance` instead ?
